torchvision/prototype/transforms/_transform.py

```diff
@@ -56,10 +56,19 @@ def extra_repr(self) -> str:
 
         return ", ".join(extra)
 
-    # This attribute should be set on all transforms that have a v1 equivalent. Doing so enables the v2 transformation
-    # to be scriptable. See `_extract_params_for_v1_transform()` and `__prepare_scriptable__` for details.
+    # This attribute should be set on all transforms that have a v1 equivalent. Doing so enables two things:
+    # 1. In case the v1 transform has a static `get_params` method, it will also be available under the same name on
+    #    the v2 transform. See `__init_subclass__` for details.
+    # 2. The v2 transform will be JIT scriptable. See `_extract_params_for_v1_transform` and `__prepare_scriptable__`
+    #    for details.
     _v1_transform_cls: Optional[Type[nn.Module]] = None
 
+    def __init_subclass__(cls) -> None:
```

Comment:
As an alternative to `__init_subclass__()`  (of which I keep forgetting the existence and the purpose), would this work too?

```py
@staticmethod
def get_params(cls):
   if cls._v1_transform_cls is not None and hasattr(cls._v1_transform_cls, "get_params"):
        return cls._v1_transform_cls.get_params()
    else:
        raise AttributeError(
            "cls {cls} has no get_params method. You probably don't need one anymore
             as the same RNG is applied to all images, bboxes and masks in the same transform call.
             If what you need is a way to transform different batches with the same RNG,
             please reach out at #1234567 (the feedback issue.
         ")

Reply:
Two problems here with JIT:

1. `get_params` takes parameters in v1, e.g. 

   https://github.com/pytorch/vision/blob/7cf0f4cc1801ff1892007c7a11f7c35d8dfb7fd0/torchvision/transforms/transforms.py#L617-L618

   Since I'm not aware of an arbitrary parameter passthrough like `(*args: Any, **kwargs: Any)` that works with JIT, we would have to implement this manually on every class that needs it.

2. You used the `@staticmethod` decorator, but take `cls` as first input. Since you need `cls`, we need to switch to `@classmethod` to make it work. That should work in eager mode, but for JIT this now means that the whole class needs to be scriptable. For example, manually defining `get_params` for `RandomCrop` as explained in 1. and trying to script it, leads to

    ```
    RuntimeError: 
    'Tensor (inferred)' object has no attribute or method '_v1_transform_cls'.:
      File "/home/philip/git/pytorch/vision/torchvision/prototype/transforms/_geometry.py", line 437
        @classmethod
        def get_params(cls, img: torch.Tensor, output_size: Tuple[int, int]) -> Tuple[int, int, int, int]:
            return cls._v1_transform_cls.get_params(img, output_size)
                   ~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    ```

   It seems it can't infer the type of `cls` and uses `torch.Tensor` as fallback. Annotation `cls: Type[RandomCrop]` yields 
    
    ```
    RuntimeError: 
    Unknown type constructor Type:
    ```

I agree using `__init_subclass__` is unconventional, but it seems like the cleanest solution here. Since we actually alias the function, we avoid all of the JIT crazyness that we would have to deal with otherwise. If you can find a working solution, I'm happy to adopt it though.

Reply:
Thanks for the details. My last try would be to declare `get_params` as a `@property` but... let me guess... JIT doesn't support it?

Reply:
You guessed right. But even if it would that wouldn't work. `@property` needs an instance.